pkg/parse/updater.go

```diff
@@ -115,54 +133,111 @@ func (u *updater) Update(ctx context.Context, cache *cacheForCommit) status.Mult
 		u.updateMux.Unlock()
 	}()
 
-	objs := filesystem.AsCoreObjects(cache.objsToApply)
-
-	// Update the declared resources so that the Remediator immediately
-	// starts enforcing the updated state.
-	if !cache.resourceDeclSetUpdated {
-		var validationErrs status.MultiError
-		objs, validationErrs = u.resources.Update(ctx, objs)
-		u.setValidationErrs(validationErrs)
-		if validationErrs != nil {
-			klog.Warningf("Failed to validate declared resources: %v", validationErrs)
-			return validationErrs
+	updateErrs := u.update(ctx, cache)
+
+	// Prepend current conflict errors
+	var errs status.MultiError
+	errs = status.Append(errs, u.conflictErrors())
+	errs = status.Append(errs, updateErrs)
+	return errs
+}
+
+// update performs most of the work for `Update`, making it easier to
+// consistently prepend the conflict errors.
+func (u *updater) update(ctx context.Context, cache *cacheForCommit) status.MultiError {
+	// Stop remediator workers.
+	// This prevents objects been updated in the wrong order (dependencies).
+	// Continue watching previously declared objects and updating the queue.
+	// Queued objects will be remediated when the workers are started again.
+	u.remediator.Pause()
+
+	// Update the declared resources (source of truth for the Remediator).
+	// After this, any objects removed from the declared resources will no
+	// longer be remediated, if they drift.
+	if !cache.declaredResourcesUpdated {
+		objs := filesystem.AsCoreObjects(cache.objsToApply)
+		_, err := u.declare(ctx, objs)
+		if err != nil {
+			return err
 		}
+		// Only mark the declared resources as updated if there were no (non-blocking) parse errors.
+		// This ensures the update will be retried until parsing fully succeeds.
```

Comment:
Thanks for adding all of these comments and breaking out the helper methods, it helps with following the control flow here